velox/functions/prestosql/ArrayFunctions.h

```diff
@@ -356,4 +356,52 @@ struct ArrayHasDuplicatesFunction {
   }
 };
 
+template <typename TExecParams, typename T>
+struct ArrayNormalizeFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(TExecParams);
+
+  FOLLY_ALWAYS_INLINE bool callNullFree(
+      out_type<velox::Array<T>>& result,
+      const null_free_arg_type<velox::Array<T>>& inputArray,
+      const null_free_arg_type<T> p) {
+    VELOX_USER_CHECK_GE(
+        p, 0, "array_normalize only supports non-negative p: {}", p);
+
+    // If the input array is empty, then the empty result should be returned.
+    if (inputArray.size() == 0) {
+      return true;
+    }
+
+    result.reserve(inputArray.size());
+
+    // If p = 0, then it is L0 norm. Presto version returns the input array.
+    if (p == 0) {
+      result.add_items(inputArray);
+      return true;
+    }
+
+    // Calculate p-norm.
+    T sum = 0;
+    for (const auto& item : inputArray) {
+      sum += pow(abs(item), p);
+    }
+
+    T pNorm = pow(sum, 1.0 / p);
+
+    // Presto version also returns the input array if p-norm = 0.
+    // Means that the array is a zero vector.
```

Comment:
Rephrase : "If the input array is a zero vector then pNorm = 0. Return the input array for this case."